### ISSUE TITLE
Cannot train data with several days， but can train success with one day data

### ISSUE THREAD

**Issue by WoNiuHu** [microsoft/LightGBM#2597]
## Environment info

Operating System:
linux
CPU/GPU model:
cpu
C++/Python/R version:
python
LightGBM version or commit hash:
2.3.0
## Error message
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] Stopped training because there are no more leaves that meet the split requirements

## Reproducible examples

# task type, support train and predict
task = train
boosting_type = gbdt
objective = binary
metric = binary_logloss,auc
metric_freq = 1
is_training_metric = true
boost_from_average = false
max_bin = 255

max_depth = 8
learning_rate = 0.4
num_leaves = 45
feature_fraction = 0.8
bagging_freq = 5
bagging_fraction = 0.8
min_data_in_leaf = 200
min_sum_hessian_in_leaf = 5.0
is_enable_sparse = true
is_save_binary_file = false
lambda_l2 = 0.001
## Steps to reproduce

1.  when I use one day data to train the model with the train.conf, it finished iteration 47 even though I set the number_trees = 128, but when I train the model with two days, or more days, it cannot split, and cannot generate any tree. 
2.
3.

**Comment by StrikerRUS** [comment]
Unfortunately, without [MCVE](https://stackoverflow.com/help/mcve) it's hard to say what's going wrong here.

I guess you should double check your data, e.g. it has the same distribution, there are no constant features, columns are not mixed for different days, and it makes sense in general.

Also, you can try to play around with params, e.g. with
```
min_data_in_leaf = 200
min_sum_hessian_in_leaf = 5.0
```

**Comment by WoNiuHu** [comment]
> Unfortunately, without [MCVE](https://stackoverflow.com/help/mcve) it's hard to say what's going wrong here.
> 
> I guess you should double check your data, e.g. it has the same distribution, there are no constant features, columns are not mixed for different days, and it makes sense in general.
> 
> Also, you can try to play around with params, e.g. with
> 
> ```
> min_data_in_leaf = 200
> min_sum_hessian_in_leaf = 5.0
> ```

thanks. The data of different days has the same distribution,  
It is said that the amount of data is not split, but the amount is small can split, but why?
